agent/serf.go

```diff
@@ -131,3 +150,49 @@ type retryingClient struct {
 	client *serf.RPCClient
 	config serf.Config
 }
+
+// MockSerfClient is a mock/fake Serf Client used in testing
+type MockSerfClient struct {
+	members []serf.Member
+	coords  map[string]*coordinate.Coordinate
+}
+
+// NewMockSerfClient is a helper function used to create a mock/fake Serf Client
+// used in testing
+func NewMockSerfClient(members []serf.Member, coords map[string]*coordinate.Coordinate) (client *MockSerfClient, err error) {
```

Comment:
We already had a chat with Roman about this and decided that it didn't make sense to create a separated test file just for this, plus it was nice to have it here since it's logically tied to the Serf Client above (we were looking for one here initially).

If you feel that's bad let me know and I'll move it somewhere else, but we felt it was best if left in here.

Reply:
Not sure I follow but an implementation which is only useful in tests (I mean the whole `MockSerfClient` type) does not really belong here.
Besides it is not really a `mock` in the sense it is used in testing - mock objects usually record exact interactions with the API (i.e. which methods and with what parameters were invoked in a test) while this is more or a test duplicate or replacement that simply hands off dummy data.

Reply:
okay, I'll move it to mock_serf.go then?

Reply:
done.